### ISSUE TITLE
borg create: No archive specified but it is specified!

### ISSUE THREAD

**Issue by fujisan43** [borgbackup/borg#1930]
Hello,

When I run the following
```
[smith@host2] $ borg create -v --info --stats --exclude '*.gvfs*' --exclude '*.thumbnails*' \
--exclude '*.backup*' --exclude '*.ccache' --exclude '*.[Cc]ache' --exclude '*/[Cc]ache' \
--exclude '*/Downloads' --exclude '*/[Tr]ash*' --exclude '*~' \
host1:/data/borgbackups/smith::2016-12-05@11:28:34 /home/smith
```
I get this error:
```
borg create: error: argument ARCHIVE: "host1:/data/borgbackups/smith::2016-12-05@09:46:28": No archive specified
```

It used to work fine before I upgraded borgbackup to 1.0.8
```
[smith@host2] $ rpm -q borgbackup
borgbackup-1.0.8-2.fc25.x86_64
```

Fuji

**Comment by ThomasWaldmann** [comment]
Try smith@host1 instead of just host1.

I suspect the parser gets confused by you using @ inside the archive name.

Possibly a duplicate of #1929.

You said it worked before - what version did you use before 1.0.8?

**Comment by fujisan43** [comment]
Yes, it's working now with smith@host1:...
It is actually the same problem as [#1929](https://github.com/borgbackup/borg/issues/1929).
I think the previous version I used was 1.0.7.

**Comment by ThomasWaldmann** [comment]
OK, maybe this was unintentionally caused by some parsing regex change.

We'll have a look if we can improve that...

**Comment by ThomasWaldmann** [comment]
This fails:
```
(borg-env)tw@tux:~/w/borg$ borg create localhost:/home/tw/w/borg/repo::2016-12-05@11:28:34 borg
usage: borg create [-h] [--critical] [--error] [--warning] [--info] [--debug]
...
borg create: error: argument ARCHIVE: "localhost:/home/tw/w/borg/repo::2016-12-05@11:28:34": No archive specified
```
This works:
```
(borg-env)tw@tux:~/w/borg$ borg create localhost:/home/tw/w/borg/repo::2016-12-05@112834 borg
```
So it seems the colons in the daytime part confused the parser.

**Comment by ThomasWaldmann** [comment]
```
    assert repr(Location('host:/path::2016-12-05@11:28:34')) == \
        "Location(proto='ssh', user=None, host='host', port=None, path='/path', archive='2016-12-05@11:28:34')", \
       "repr: %r" % Location('host:/path::2016-12-05@11:28:34')

E   AssertionError: repr: Location(proto='ssh', user='host:/path::2016-12-05', host='11', port=None, path='28:34', archive=None)
```

**Comment by fujisan43** [comment]
I don't have trouble with the columns:

```
$ borg create -v --info --stats smith@host:/data/borgbackups/smith::2016-12-09@11:04:39 /home/smith
------------------------------------------------------------------------------
Archive name: 2016-12-09@11:04:39
Archive fingerprint: 65ff26df61d46496ab1eecc69e4fabab81cd387f760d27389c3275ba952ebb2c
Time (start): Fri, 2016-12-09 11:05:43
Time (end):   Fri, 2016-12-09 11:06:10
Duration: 26.83 seconds
Number of files: 3360
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:              815.76 MB            815.77 MB             67.07 MB
All archives:                1.57 GB              1.57 GB            823.73 MB

                       Unique chunks         Total chunks
Chunk index:                    3621                 4286
------------------------------------------------------------------------------
$
$
$
$ borg list smith@host:/data/borgbackups/smith
2016-11-04@10:26:24                  Fri, 2016-11-04 10:26:27
2016-12-09@11:04:39                  Fri, 2016-12-09 11:05:43

```

**Comment by ThomasWaldmann** [comment]
@fujisan43 It was the "@", I fixed my comment above.

**Comment by ThomasWaldmann** [comment]
fixed by #1937.

**Comment by tuxick** [comment]
Actually it's not the @ itself, it's correct if preceded by a username. Had a script failing to do that :)
